docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,981 @@
+        mygc = ["immortalspace", "largeobjectspace"]
+        ```
+        This adds a build-time flag for `mygc`, and tells the compiler that 
+        `mygc` will use an immortal space and large object space.
+    2. `mmtk-core/src/plan/mod.rs`, under the import statements, add:
+        ```rust
+        #[cfg(feature = "mygc")]
+        pub mod mygc;
+        #[cfg(feature = "mygc")]
+        pub use self::mygc as selected_plan;
+        ```
+        This adds `mygc` as a module, which can be conditionally compiled 
+        using the feature (or environment variable) `mygc`. A GC plan needs 
+        to be selected at build time, and only one plan can be selected 
+        (as `selected_plan`).
+    3. `mmtk-openjdk/mmtk/Cargo.toml`, under `[features]`, add: 
+        ```rust 
+        mygc = ["mmtk/mygc"] 
+        ```
+        This adds the build flag to the binding crate, using the `mygc` 
+        flag from mmtk-core.
+    
+Note that all of the above changes almost exactly copy the NoGC entries in 
+each of these files. However, NoGC has some variants, such as a lock-free 
+variant, that are not needed for this tutorial. Remove references to them in 
+the MyGC plan now. 
+1. Within `mygc/global.rs`, find any use of `#[cfg(feature = "mygc_lock_free")]` 
+and delete both it *and the line below it*.
+2. Then, delete any use of the above line's negation, 
+`#[cfg(not(feature = "mygc_lock_free"))]`, this time without changing the 
+line below it.
+
+After you rebuild OpenJDK (and the MMTk core), you can use MyGC. Try testing it 
+with the each of the three benchmarks. It should work identically to NoGC.
+
+At this point, you should familiarise yourself with the MyGC plan if you 
+haven't already. Try answering the following questions by looking at the code 
+and [Further Reading](#further-reading): 
+   * Where is the allocator defined?
+   * How many memory spaces are there?
+   * What kind of memory space policy is used?
+   * What happens if garbage has to be collected?   
+
+[**Back to table of contents**](#contents)
+
+
+
+***
+## Building a Semispace Collector
+### What is a Semispace collector?
+In a Semispace collector, the heap is divided into two equally-sized spaces, 
+called 'semispaces'. One of these is defined as a 'fromspace', and the other 
+a 'tospace'. The allocator allocates to the tospace until it is full. 
+
+When the tospace is full, a stop-the-world GC is triggered. The mutator is 
+paused, and the definitions of the spaces are flipped (the 'tospace' becomes 
+a 'fromspace', and vise versa). Then, the collector scans each object in what 
+is now the fromspace. If a live object is found, a copy of it is made in the 
+tospace. That is to say, live objects are copied *from* the fromspace *to* 
+the tospace. After every object is scanned, the fromspace is cleared. The GC 
+finishes, and the mutator is resumed.
+
+### Allocation: Add copyspaces
+
+The first step of changing the MyGC plan into a Semispace plan is to add the 
```

Comment:
~changing the MyGC plan into a Semispace plan~ changing your MyGC plan from one that cannot collect garbage into one that implements the semispace algorithm